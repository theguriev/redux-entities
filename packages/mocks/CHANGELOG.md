# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 1.11.19 (2021-09-07)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.18 (2021-09-01)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.17 (2021-09-01)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.16 (2021-08-30)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.15 (2021-08-30)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.14 (2021-08-27)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.13 (2021-08-25)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.12 (2021-08-24)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.11 (2021-08-20)

**Note:** Version bump only for package @redux-things/mocks





## 1.11.10 (2021-08-20)

**Note:** Version bump only for package @redux-things/mocks





# [1.10.0](https://github.com/theguriev/redux-things/compare/v1.9.0...v1.10.0) (2021-06-21)


### Features

* **dumb:** init ([a7691a1](https://github.com/theguriev/redux-things/commit/a7691a1c0529e4dd4cf27d3154b9984e1100e2eb))
* **dumb:** working on dependencies ([a08e36a](https://github.com/theguriev/redux-things/commit/a08e36a25c3b83a8127d335312fea93ee86ff57f))





# [1.7.0](https://github.com/theguriev/redux-things/compare/v1.6.2...v1.7.0) (2021-06-01)


### Features

* **core:** add actions to returns ([999a9c2](https://github.com/theguriev/redux-things/commit/999a9c2975c7069b6d896aa1172dc64eb6e781a9))





## [1.5.2](https://github.com/theguriev/redux-things/compare/v1.5.1...v1.5.2) (2021-05-24)

**Note:** Version bump only for package @redux-things/mocks





## [1.5.1](https://github.com/theguriev/redux-things/compare/v1.5.0...v1.5.1) (2021-05-24)

**Note:** Version bump only for package @redux-things/mocks





# [1.5.0](https://github.com/theguriev/redux-things/compare/v1.4.4...v1.5.0) (2021-05-24)

**Note:** Version bump only for package @redux-things/mocks
