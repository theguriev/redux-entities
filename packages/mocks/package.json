{
    "name": "@redux-things/mocks",
    "version": "1.5.0",
    "description": "A library that allows you to quickly and easily synchronize your store with the server.",
    "keywords": [
        "redux",
        "queries",
        "things",
        "data",
        "data management",
        "caching",
        "cache",
        "fetch",
        "promise",
        "store",
        "state"
    ],
    "main": "dist/index.js",
    "scripts": {
        "build": "node build.js",
        "watch": "watch 'npm run build' src",
        "lint": "npx eslint --fix --ignore-path ../../.gitignore ."
    },
    "author": "Eugen Guriev",
    "license": "MIT",
    "publishConfig": {
        "access": "public"
    },
    "babel": {
        "plugins": [
            [
                "module-resolver",
                {
                    "root": [
                        "./src"
                    ],
                    "alias": {
                        "@": "./src"
                    }
                }
            ],
            "dynamic-import-node",
            "add-module-exports"
        ],
        "presets": [
            [
                "@babel/preset-env",
                {
                    "targets": {
                        "chrome": 78,
                        "edge": 15,
                        "esmodules": false
                    }
                }
            ],
            "@babel/preset-react"
        ]
    },
    "dependencies": {
        "@redux-things/core": "^1.5.0",
        "@reduxjs/toolkit": "^1.5.0",
        "core-js": "^2.6.12",
        "object-hash": "^2.1.1"
    },
    "devDependencies": {
        "@babel/cli": "^7.11.6",
        "@babel/core": "^7.11.6",
        "@babel/plugin-transform-modules-commonjs": "^7.13.8",
        "@babel/preset-env": "^7.11.5",
        "@babel/preset-react": "^7.10.4",
        "@storybook/addon-actions": "^6.1.21",
        "@storybook/addon-essentials": "^6.1.21",
        "@storybook/addon-links": "^6.1.21",
        "@storybook/react": "^6.1.21",
        "@svgr/webpack": "^5.4.0",
        "@testing-library/react-hooks": "^5.1.2",
        "babel-jest": "^26.6.3",
        "babel-loader": "^8.1.0",
        "babel-plugin-add-module-exports": "^1.0.4",
        "babel-plugin-dynamic-import-node": "^2.3.3",
        "babel-plugin-module-resolver": "^4.0.0",
        "clean-webpack-plugin": "^3.0.0",
        "copy-dir": "^1.3.0",
        "eslint": "^7.26.0",
        "eslint-config-airbnb": "^18.2.1",
        "eslint-plugin-import": "^2.23.2",
        "eslint-plugin-jest": "^24.3.6",
        "eslint-plugin-jsx-a11y": "^6.4.1",
        "eslint-plugin-react": "^7.23.2",
        "jest": "^26.6.3",
        "ts-jest": "^26.5.5",
        "watch": "^1.0.2"
    },
    "peerDependencies": {
        "object-hash": "^2.1.1",
        "prop-types": "^15.7.2",
        "react": "^17.0.0",
        "react-dom": "^17.0.0",
        "react-redux": "^7.2.2",
        "react-use": "^17.1.1"
    },
    "bugs": {
        "url": "https://github.com/theguriev/redux-things/issues"
    },
    "homepage": "https://github.com/theguriev/redux-things",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/theguriev/redux-things.git"
    },
    "gitHead": "046bdecff9e5080cdd59c39aac61e3052588f3e1"
}
