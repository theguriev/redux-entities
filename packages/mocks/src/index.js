export * from './reduxContextProvider'
export * from './useThing'
export * from './launchFlow'
