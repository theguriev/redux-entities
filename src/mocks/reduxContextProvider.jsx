import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { setupAsyncReducers } from '@/hooks'

const store = setupAsyncReducers(
    createStore(v => v),
    {}
)

export const reduxContextProvider = ({ children }) => <Provider store={store}>{children}</Provider>
