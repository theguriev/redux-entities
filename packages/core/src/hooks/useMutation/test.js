import { renderHook, act } from '@testing-library/react-hooks'
import { reduxContextProvider as wrapper } from '@redux-things/mocks'
import { useMutation } from '.'

describe('useMutation', () => {
    test('basic mutation', async () => {
        const { result, waitForValueToChange } = renderHook(
            () => useMutation(
                ({ options }) => Promise.resolve(`Here we will send some data to the server ðŸ˜‡ ${options.test}`)
            ),
            { wrapper }
        )
        act(() => {
            result.current.mutate({ test: 'TEST OPTIONS' })
        })
        await waitForValueToChange(() => result.current.isLoading)
        expect(result.current.data).toBe('Here we will send some data to the server ðŸ˜‡ TEST OPTIONS')
        expect(result.all).toMatchSnapshot()
    })
})
