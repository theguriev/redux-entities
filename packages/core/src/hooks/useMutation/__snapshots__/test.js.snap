// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMutation basic mutation 1`] = `
Array [
  Object {
    "data": null,
    "error": null,
    "isError": false,
    "isLoading": false,
    "mutate": [Function],
    "toType": [Function],
  },
  Object {
    "data": null,
    "error": null,
    "isError": false,
    "isLoading": true,
    "mutate": [Function],
    "toType": [Function],
  },
  Object {
    "data": "Here we will send some data to the server ðŸ˜‡ TEST OPTIONS",
    "error": null,
    "isError": false,
    "isLoading": false,
    "mutate": [Function],
    "toType": [Function],
  },
]
`;
